### PR TITLE
Bottles now use dynamic reagent filling sprites, tweaks to reagent container sprites

### LINKED ISSUE #3881: [Spriting] Large Bluespace Beakers still don't have a sprite for the lid
I personally hold @skowron accountable because i mentioned this in passing to him once several months ago.  However I think its time to get on it and make a proper issue.  No lid no peace.

image

### PR DESCRIPTION
**Pull request by ghost**
Progress in our code !
- Fix #3881 for real this time. Lids and beaker sprites were all over the place (normal beakers, bottles and vials were offset one on the left, bluespace beakers were centered but offset two up, etc). Now all sprites expect for noreact are centered horizontally and synced vertically

Now onto the main change :
- Bottles now use dynamic reagent fillings. For those who don't know what those are, when you fill a beaker with acid it very clearly shows the volume and the acid's color (orange). Bottles didn't have this, until now
- Due to the way this works, a standard bottle sprite has replaced the dozens of snowflake bottles with pretty pre-filled sprites. The standard bottle sprite uses the same colors as the beaker
- Bottles now fill dynamically every five units. Rounding used is +/- 2.5 % per step rounded up
- Now, you will be able to tell immediately which bottle contains plasma, which one contains acid and which one contains your precious chloral
- All pre-spawned bottles now use the base sprite. The reagent contents are still fairly obvious (especially for odd ones like ammonia). The custom sprites are still in the code but now no longer used (theorically)
- Standardized procs from beaker to make sure sprite updates correctly

In-game example :

scrnshot1

<s>Only current problem is that the Chemmaster had to be jury-rigged to not let the user use custom sprites, which are now outdated and most likely completely deprecated due to reagent fillings. Thus there is a "broken image" link replacing what used to be the bottle sprite selector. All code related to bottle sprite picking was commented for the time being</s>

### CODE REVIEW

**Comment by 9600bauds** [comment]
But
Muh
Skinzzzz

**Comment by dylanstrategie** [comment]
It's extremely hard to not make the bottle look squarish without it looking weird

As for the Chemmaster UI, I honestly don't know how to remove the broken button. I'm not good at all with UI code

**Comment by smet19** [comment]
It would be better if you add 3 different bottle sprites
bottle
bottle2
bottle3

**Comment by ComicIronic** [comment]
Fix the UI

**Comment by dylanstrategie** [comment]
Jawohl

**Comment by dylanstrategie** [comment]
What I think could happen smet is having a very few bottle shapes (note that they all need custom lids and reagent fillings, so it's more work than it looks) and then assigning this to pre-existing bottles as needed. Although the first and the third are strikingly look-alike, and the second is a perfume bottle

I'm also considering doing the same things for pills, color depends on pill content and there's a few basic pill shapes

**Comment by 9600bauds** [comment]
Don't you dare killing color-coded pills.
Since when are pills transparent anyways?

File: code/modules/reagents/reagent_containers/glass/bottle.dm [hunk 1]
```diff
@@ -29,212 +31,264 @@
...
 	desc = "A small bottle. Contains liver enhancement virion culture in synthblood medium."
 	icon = 'icons/obj/chemical.dmi'
-	icon_state = "bottle3"
+	//icon_state = "bottle3"
 	New()
 		..()
 		var/datum/disease/F = new /datum/disease/advance/heal(0)
 		var/list/data = list("viruses"= list(F))
 		reagents.add_reagent("blood", 20, data)
+		update_icon()
```
Comment (ComicIronic) [inline code/modules/reagents/reagent_containers/glass/bottle.dm]: Don't do this.

You should just include update_icon() in on_reagent_change() for bottles.
It's how beakers do it.

Comment (dylanstrategie) [inline code/modules/reagents/reagent_containers/glass/bottle.dm]: Beakers do the same thing for all their pre-defined types though. Should I change that too ?

Comment (ComicIronic) [inline code/modules/reagents/reagent_containers/glass/bottle.dm]: It should work without.
Don't stretch the PR into beakers, just don't include them here.
Did you test without?

Comment (dylanstrategie) [inline code/modules/reagents/reagent_containers/glass/bottle.dm]: I'll try to standardize the New() proc

**Comment by smet19** [comment]
@dylanstrategie
I only dropped some examples, you can do what ever you want

**Comment by dylanstrategie** [comment]
Changelog at #5106

**Pull request merged**